lib/src/tokenizer.cc

```diff
@@ -22,7 +22,7 @@ namespace hocon {
     /** Whitespace Saver */
     token_iterator::whitespace_saver::whitespace_saver() : _last_token_was_simple_value(false) { }
 
-    void token_iterator::whitespace_saver::add(char c) {
+    void token_iterator::whitespace_saver::add(signed char c) {
```

Comment:
This should really be reworked to use `istream` directly, it already supports the idea of `putback`. However, that can be left as a separate ticket if we need to get things building.